torch/_dynamo/variables/misc.py

```diff
@@ -714,6 +714,43 @@ def call_function(
             return variables.ListIteratorVariable(
                 items, mutable_local=MutableLocal(), **options
             )
+        elif self.value is itertools.accumulate and not kwargs:
+            from .builtin import BuiltinVariable
+
+            if len(args) == 1 and args[0].has_unpack_var_sequence(tx):
+                seq = args[0].unpack_var_sequence(tx)
+
+                def func(tx, item, acc):
+                    return BuiltinVariable(sum).call_function(tx, [item, acc], {})
+
+            elif (
+                len(args) == 2
+                and args[0].has_unpack_var_sequence(tx)
+                and args[1].is_callable(tx)
+            ):
+                seq = args[0].unpack_var_sequence(tx)
+                func = args[1].call_function
+
+            else:
+                raise unimplemented("Unsupported arguments for itertools.accumulate")
+
+            items = []
+            acc = None
+            for item in seq:
+                if acc is None:
+                    acc = item
+                else:
+                    try:
+                        acc = func(tx, item, acc)
+                    except Exception:
+                        raise unimplemented(
+                            f"Unexpected failure in invoking function during accumulate. Failed running func {func}({item}{acc})"
+                        )
```

Comment:
```suggestion
                    except Exception as ex:
                        raise unimplemented(
                            f"Unexpected failure in invoking function during accumulate. Failed running func {func}({item}{acc})"
                        ) from ex

Reply:
Catching a bare exception is really frowned upon in general here though, I would recommend trying to limit what excetion you are catching signficantly. Is it a RuntimeException? and UnimplementedException etc...